url2698a2c731_cassandra_project_cassandra_stand_alone_tgz-pJ8-eu_cassandra_sim_MySimpleDynamicScenarioSimulationJ8/src/eu/cassandra/sim/math/response/PricingVector.java

```diff
@@ -0,0 +1,268 @@
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package eu.cassandra.sim.math.response;
+
+import java.util.ArrayList;
+
+import eu.cassandra.sim.utilities.Constants;
+
+/**
+ * This class is used for implementing an array of pricings that are a result
+ * of the analysis of the available pricing schemes (Basic and New). The array
+ * is then used as an input to the Response Models creation procedure.
+ * 
+ * @author Antonios Chrysopoulos
+ * @version 0.9, Date: 29.07.2013
+ */
+public class PricingVector
+{
+
+  /** This is an list of the available pricings for the day. */
+  ArrayList<Pricing> pricings = new ArrayList<Pricing>();
+
+  /** This is an list of the the penalty indices in the pricing list. */
+  ArrayList<Integer> penalties = new ArrayList<Integer>();
+
+  /** This is an list of the the penalty indices in the pricing list. */
+  ArrayList<Integer> rewards = new ArrayList<Integer>();
+
+  /** This is an list of the the penalty indices in the pricing list. */
+  ArrayList<Integer> bases = new ArrayList<Integer>();
+
+  /** This variable shows the index to the cheapest pricing of the list. */
+  ArrayList<Integer> indexOfCheapest = new ArrayList<Integer>();
+
+  /**
+   * The constructor of the Pricing Vector. It uses the pricing schemes to
+   * fill the array and the rest of the variables.
+   * 
+   * @param basicScheme
+   *          The basic pricing scheme as imported by the user to the Training
+   *          Module GUI.
+   * @param newScheme
+   *          The new pricing scheme as imported by the user to the Training
+   *          Module GUI.
+   */
+  public PricingVector (double[] basicScheme, double[] newScheme)
+  {
+    // Initializing auxiliary variables
+    boolean startFlag = false;
+    int start = -1;
+    int end = -1;
+    String type = "";
+    double currentValue = 0;
+    double previousPricing = 0;
+    double previousValue = 0;
+
+    // For all the minutes of the day the procedure checks the difference
+    // between the two pricing schemes and defines the incentives found.
+    for (int i = 0; i < newScheme.length; i++) {
+      // System.out.println("Index:" + i);
+
+      if (previousValue != newScheme[i]) {
+        // System.out
+        // .println("In for difference!" + previousValue + " " + diff[i]);
+
+        if (startFlag == false) {
+          // System.out.println("In for start!");
+          currentValue = newScheme[i];
+          previousPricing = basicScheme[i];
+          if (newScheme[i] == basicScheme[i])
+            type = "Base";
+          else if (newScheme[i] < basicScheme[i])
+            type = "Reward";
+          else
+            type = "Penalty";
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 84:89, 104:109.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.